src/id3/v2/tag.rs

```diff
@@ -320,8 +331,18 @@ impl Id3v2Tag {
 	}
 
 	/// Removes a [`Frame`] by id
-	pub fn remove(&mut self, id: &str) {
-		self.frames.retain(|f| f.id_str() != id)
+	pub fn remove(&mut self, id: &str) -> impl Iterator<Item = Frame<'static>> + '_ {
```

Comment:
Passing a `FrameId<'_>` would be type safe and avoid the somehow unrelated `eq_ignore_ascii_case()`. Otherwise we should extract a function that explcitly matches `FrameId` with `str` consistently instead of inlining this code here and probably elsewhere.

Reply:
Yeah, I'll change that soon in a follow up PR. I don't know how many places it's used right now.